test/integration/roundtrip_test.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+
+	// metav1 "k8s.io/kubernetes/pkg/apis/meta/v1"
+
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	genericserveroptions "k8s.io/kubernetes/pkg/genericapiserver/options"
+
+	// TODO: fix this upstream
+	// we shouldn't have to install things to use our own generated client.
+
+	// avoid error `servicecatalog/v1alpha1 is not enabled`
+	_ "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/install"
+	// avoid error `no kind is registered for the type v1.ListOptions`
+	_ "k8s.io/kubernetes/pkg/api/install"
+
+	// to start our server locally
+	"github.com/kubernetes-incubator/service-catalog/cmd/apiserver/app/server"
+	// our versioned types
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
+	// our versioned client
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	servicecatalogclient "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+)
+
+// TestGroupVersion is trivial.
+func TestGroupVersion(t *testing.T) {
+	client, shutdownServer := getFreshApiserverAndClient(t)
+	defer shutdownServer()
+
+	gv := client.Servicecatalog().RESTClient().APIVersion()
+	if gv.Group != servicecatalog.GroupName {
+		t.Fatal("we should be testing the servicecatalog group, not ", gv.Group)
+	}
+}
+
+// TestNoName checks that all creates fail for objects that have no
+// name given.
+func TestNoName(t *testing.T) {
+	client, shutdownServer := getFreshApiserverAndClient(t)
+	defer shutdownServer()
+	scClient := client.Servicecatalog()
+
+	ns := "namespace"
+
+	br, e := scClient.Brokers().Create(&v1alpha1.Broker{})
+	if nil == e {
```

Comment:
As a general comment throughout the tests, see if you can merge the if statements with the statements before them. So this would be:

```
if br, e := scClient.Brokers().Create(&v1alpha1.Broker{}); nil == e {
    ...
}
```

All these variables seem to only be used in the scope of erroring out.